File: backend/app/schemas/analytics.py
from pydantic import BaseModel, ConfigDict


class AnalyticsOverviewRead(BaseModel):
    total_tickets: int
    open_tickets: int
    resolved_tickets: int
    urgent_tickets: int
    ai_suggestions_count: int
    ai_approved_count: int
    ai_adoption_rate: float
    category_distribution: dict[str, int]
    priority_distribution: dict[str, int]

    model_config = ConfigDict(extra="forbid")

File: frontend/src/api/analytics.ts
import { apiClient } from "./client";

export type AnalyticsOverviewRead = {
  total_tickets: number;
  open_tickets: number;
  resolved_tickets: number;
  urgent_tickets: number;
  ai_suggestions_count: number;
  ai_approved_count: number;
  ai_adoption_rate: number;
  category_distribution: Record<string, number>;
  priority_distribution: Record<string, number>;
};

export async function getAnalyticsOverview() {
  const response = await apiClient.get<AnalyticsOverviewRead>("/analytics/overview");
  return response.data;
}
